### PR TITLE
hotfix: rp panics due to"subt" nil pointer

### PR DESCRIPTION
**Pull request by Nasfame**

### CODE REVIEW

**Comment by lukemarsden** [comment]
surely this would panic in the same way?

nothing in this change checks for a nil, and the panic is a nil pointer deref

image

**Comment by lukemarsden** [comment]
i mean, i see you've added a different spelling to the function list, so maybe it works around it, but we should really make the RP resilient to panics in the template parsing code entirely. Let's do that in this change, please

**Comment by Nasfame** [comment]
but what was happening instead was parsed 1st then list the funcs

And i added a new alias that's it

**Comment by lukemarsden** [comment]
ok, but what if user passes function = `sdflakjs`? would it panic then?

**Comment by Nasfame** [comment]
@lukemarsden interesting,  ok i ill add a recover logic in that case

**Comment by Nasfame** [comment]
Done @lukemarsden

**Comment by lukemarsden** [comment]
What is the user facing behaviour in the panic case?

**Comment by Nasfame** [comment]
I m installing tests in place

**Comment by Nasfame** [comment]
> What is the user facing behaviour in the panic case?
Panics for rp is recovered. Just error

This is a simple reproducer: https://go.dev/play/p/MWo0vRNPnqr

**Comment by Nasfame** [comment]
@lukemarsden this is the fix

File: pkg/module/utils.go [hunk 1]
```diff
@@ -231,13 +243,16 @@ func LoadModule(module data.ModuleConfig, inputs map[string]string) (*data.Modul
...
 		if err != nil {
+			log.Debug().Err(err).Msgf("failed to parse inputs")
 			return nil, fmt.Errorf("unable to marshal string %q", v)
 		}
 		newInputs[k] = string(bs)
 	}
 
 	var template bytes.Buffer
 	if err := tmpl.Execute(&template, newInputs); err != nil {
+		log.Debug().Err(err).Msgf("failed to executue template")
```
Comment (lukemarsden) [inline pkg/module/utils.go:254]: typo execute

File: pkg/module/utils.go [hunk 2]
```diff
@@ -249,6 +264,7 @@ func LoadModule(module data.ModuleConfig, inputs map[string]string) (*data.Modul
 	var moduleData data.Module
 	bs := template.Bytes()
 	if err := json.Unmarshal(bs, &moduleData); err != nil {
+		log.Debug().Err(err).Msgf("failed to unmarshall module")
```
Comment (lukemarsden) [inline pkg/module/utils.go:267]: typo unmarshal

File: pkg/module/utils.go [hunk 3]
```diff
@@ -211,18 +211,30 @@ func subst(format string, jsonEncodedInputs ...string) string {
 // - inject the given values using template syntax
 // - JSON parse and check we don't have errors
 func LoadModule(module data.ModuleConfig, inputs map[string]string) (*data.Module, error) {
+	defer func() {
+		a := recover()
+		if a != nil {
+			log.Error().Any("panic", a).Msgf("panic handler")
```
Comment (lukemarsden) [inline pkg/module/utils.go:217]: maybe make the message say "recovered panic in LoadModule" or is that obvious from the log?

File: pkg/module/utils.go (hunk 1 above)
Comment (Nasfame) [reply pkg/module/utils.go:254]: ok

**Comment by lukemarsden** [comment]
cool, thanks, please merge :)

**Comment by Nasfame** [comment]
> What is the user facing behaviour in the panic case?

@lukemarsden i m unable to make it panic: 

image

WHen function is undefined; it returns undefined function. I think if the resulting template is invalid then it panics

**Comment by Nasfame** [comment]
Nope that doesn't panic either: 

`
image

**Review by Nasfame — COMMENTED** [review]
@lukemarsden pr is now ready

**Pull request merged**